src/util.ts

```diff
@@ -56,12 +59,46 @@ export async function zipDir(
     });
     archive.pipe(output);
     // Add dir to root of archive
-    archive.directory(dirPath, false);
+    getFiles(dirPath).forEach((filepath) => {
```

Comment:
Did you see that `archive.directory()` can take a third argument, which as far as I can see is a callback that can be used for some kind of filtering? https://github.com/archiverjs/node-archiver/blob/f6c9cc2fa6a167594c82428faf22dc59dbba7404/lib/core.js#L658-L661

Reply:
Hi @voxpelli, thanks for the review.
I tried to do this with just archiver but found that all filtering done is via [minimatch](https://github.com/isaacs/minimatch) which did not work due to https://github.com/isaacs/minimatch/issues/46. So for example, this test was failing https://github.com/google-github-actions/deploy-cloud-functions/blob/support-gcloudignore/tests/test-func-ignore-node/.gcloudignore